tokio-fs/src/file/open_options.rs

```diff
@@ -0,0 +1,209 @@
+    ///
+    /// [`custom_flags`]: https://doc.rust-lang.org/beta/std/os/unix/fs/trait.OpenOptionsExt.html#tymethod.custom_flags
+    #[cfg(unix)]
+    pub fn custom_flags(&mut self, flags: i32) -> &mut OpenOptions {
+        self.0.custom_flags(flags);
+        self
+    }
+
+    /// See the underlying [`access_mode`] call for details.
+    ///
+    /// [`access_mode`]: https://doc.rust-lang.org/beta/std/os/windows/fs/trait.OpenOptionsExt.html#tymethod.access_mode
+    #[cfg(windows)]
+    pub fn access_mode(&mut self, access: u32) -> &mut OpenOptions {
+        self.0.access_mode(access);
+        self
+    }
+
+    /// See the underlying [`share_mode`] call for details.
+    ///
+    /// [`share_mode`]: https://doc.rust-lang.org/beta/std/os/windows/fs/trait.OpenOptionsExt.html#tymethod.share_mode
+    #[cfg(windows)]
+    pub fn share_mode(&mut self, share: u32) -> &mut OpenOptions {
+        self.0.share_mode(share);
+        self
+    }
+
+    /// See the underlying [`custom_flags`] call for details.
+    ///
+    /// [`custom_flags`]: https://doc.rust-lang.org/beta/std/os/windows/fs/trait.OpenOptionsExt.html#tymethod.custom_flags
+    #[cfg(windows)]
+    pub fn custom_flags(&mut self, flags: u32) -> &mut OpenOptions {
+        self.0.custom_flags(flags);
+        self
+    }
+
+    /// See the underlying [`attributes`] call for details.
+    ///
+    /// [`attributes`]: https://doc.rust-lang.org/beta/std/os/windows/fs/trait.OpenOptionsExt.html#tymethod.attributes
+    #[cfg(windows)]
+    pub fn attributes(&mut self, attributes: u32) -> &mut OpenOptions {
+        self.0.attributes(attributes);
+        self
+    }
+
+    /// See the underlying [`security_qos_flags`] call for details.
+    ///
+    /// [`security_qos_flags`]: https://doc.rust-lang.org/beta/std/os/windows/fs/trait.OpenOptionsExt.html#tymethod.security_qos_flags
+    #[cfg(windows)]
+    pub fn security_qos_flags(&mut self, flags: u32) -> &mut OpenOptions {
+        self.0.security_qos_flags(flags);
+        self
+    }
+
+    /// Opens a file at `path` with the options specified by `self`.
+    ///
+    /// # Errors
+    ///
+    /// `OpenOptionsFuture` results in an error if called from outside of the
+    /// Tokio runtime or if the underlying [`open`] call results in an error.
+    ///
+    /// [`open`]: https://doc.rust-lang.org/std/fs/struct.OpenOptions.html#method.open
+    pub fn open<P>(&self, path: P) -> OpenOptionsFuture<P>
+    where P: AsRef<Path> + Send + 'static
+    {
+        self.clone().into_open(path)
+    }
+
+    /// Opens a file at `path` with the options specified by `self`, consuming
+    /// `self`.
+    ///
+    /// # Errors
+    ///
+    /// `OpenOptionsFuture` results in an error if called from outside of the
+    /// Tokio runtime or if the underlying [`open`] call results in an error.
+    ///
+    /// [`open`]: https://doc.rust-lang.org/std/fs/struct.OpenOptions.html#method.open
+    pub fn into_open<P>(self, path: P) -> OpenOptionsFuture<P>
```

Comment:
I didn't see this function on the `std` version. I believe this is here to avoid the clone, but I would recommend leaving this off for now.

We can always add it later if it is needed.